moped-editor/src/views/projects/projectView/ProjectComponents/utils/map.js

```diff
@@ -170,3 +172,32 @@ export const useSubcomponentOptions = (component) =>
 
     return options;
   }, [component]);
+
+/**
+ * Use Mapbox fitBounds to zoom to existing project components feature collection
+ * @param {Object} mapRef - React ref that stores the Mapbox map instance (mapRef.current)
+ * @param {Object} data - Data returned from the moped_components query
+ * @param {Array} data.project_geography - Array of existing component features
+ */
+export const useZoomToExistingComponents = (mapRef, data) => {
+  const [hasMapZoomedInitially, setHasMapZoomedInitially] = useState(false);
+
+  useEffect(() => {
+    if (!data || hasMapZoomedInitially) return;
```

Comment:
you might want to add `mapRef` to this check and return if we don't have a `current` value? in my testing the value is never undefined but i would add it mostly b/c i don't fully understand refs.

Reply:
> the Mapbox instance is assigned to mapRef.current on first render since it is passed as its ref prop

thanks for that! sounds good, but does seems like an extreme edge case unless we were caching queries.